### ISSUE TITLE
Send Later scheduled drafts not being delivered, even though scheduled send time is properly displayed in UI

### ISSUE THREAD

**Issue by atscomms** [Extended-Thunder/send-later#527]
Send Later V 9.2.7 w/ TB 102.8.0 (32-bit) on a Windows 10 64bit PC

Emails do not go out at the specified time, and they stay in Drafts folder.
The date & time appear in the Draft Folder's first column, and there is a numbered flag on the Send Later icon in the menu bar, way past the specified time.

**Comment by roorco** [comment]
Same for me,
what it is actually working is the option to use "Send" as send x miuntes later.
Other options are all broken for me, can't plan email to go out at specific time, the option grey out the send button and the mail doesn't move in any folder, I can't even edit it.

**Comment by DRSAgile** [comment]
I have TB 110, meaning not a release, but a beta version.

I do not have access to the extension's Preferences at all, there is no such a tab.

And when I choose "Send later" in a new letter menu, it goes to "Outbox" folder, but at no time I am asked or being able to set when the letter should be actually sent.

If the extension does not work with TB beta, its metafile should say accordingly, and not work in the application at all.

**Comment by firefoxlover** [comment]
I got a strange bug. I guess my drafts are stored PGP encrypted, so they reached a person without any PGP knowledge (sigh) and they couldnt read it.

Also it arrived 1h too late. I will open new issues and when I have time try to help

**Comment by kiwiheretic** [comment]
I have TB 102.7.2 (64-bit) and I also have the same problem.  The send later has the write date but the wrong number of days.

![Send Later Dialog](https://images.kiwiheretic.xyz/send-later-bug.png)

**Comment by JamesKommaderBond** [comment]
![Screen Shot 03-23-23 at 01 24 PM](https://user-images.githubusercontent.com/128706197/227204292-a1ddbad5-43d0-4a8c-91ce-906e4db547a7.JPG)


and how the hell do I get rid if that "message"?

**Comment by RichardPinka** [comment]
Same for me. 
When i started app, i simply just wanted to get send emails with 10 minutes delay (nothing else). 
but application started to inform me , that send Later : Folder Drafts  may be corrupt and i have to open its properties and repair it.

**Comment by AGI-chandler** [comment]
Add-on is completely borked for me too for unknown reasons, just started happening today for some reason.  TB 102.13.0 and SL 9.2.9.  Tried completely removing add-on and repairing and compacting Drafts folder, didn't help.  When Sending Later, messages are placed into the Drafts folder as expected, but there is no longer a "Send Later" column, not even in the list of available columns, and messages are not sent.  Maybe there are entries in about:config that need to be manually cleared?

**Comment by jikamens** [comment]
I think there may be multiple issues mixed together in this bug.

@Rings4 is correct that there are issues with Drafts folders with more than 100 messages in them. The official bug where we're tracking that issue is #529. We have a fix but it hasn't been released yet.

@roorco I'm bewildered by your description of your issue. I can't quite make out what you're saying is broken, and I don't recognize your description as something I've seen reported by others. In any case your issue sounds like it's clearly something different than the issue reported by the original author of this issue, so I think you should open a separate issue for it.

@DRSAgile The decision to list Send Later as being compatible with the TB110 beta was made by the previous maintainer. I have not investigated whether it is indeed compatible, but from your description I would assume not. In any case it's moot since it's note listed as compatible with TB115 which has now been released. We're working on the TB115 version but it's going to take a bit of time because there's some core Thunderbird functionality that needs to be added before we can update Send Later. In any case, again, this is different from the issue reported by the original author of this issue, so your comment doesn't really belong here. Let's try to keep separate issues separate, please.

@RichardPinka if the add-on says that your Drafts folder may be corrupt, then... your drafts folder may be corrupt. I'm not sure there's anything we can do about that. There is one issue we've identified in Send Later which _may_ be causing this; we will know better whether that's the case after the next Send Later release. But, again, this is not the issue reported by the original author of this issue so you should have opened a separate issue for it rather than commenting here.

@AGI-chandler I think the fact that the downgrade to 9.2.8 fixed your issue is probably a coincidence. There are some race conditions in the Send Later code which we're working on addressing which cause Send Later to fail to initialize properly for some people and not others. I don't think these are new in 9.2.9, but it's possible that other changes in the code between 9.2.8 and 9.2.9 changed the timing for you, such that the race conditions happen for you with the 9.2.9 code and note the 9.2.8 code. We're hoping to address these race conditions, or at least to try, in the next Send Later release, so hopefully it'll be stable for you again. In any case, again, this is not the issue reported by the author of this issue, so it should have been reported in a separate issue, not a comment on this issue.

**Comment by jikamens** [comment]
I believe that all of the issues flagged in this bug are resolved as of Send Later 9.3.0 (for TB109) and 10.0.1 (for TB115).

Please reopen the issue or comment if you're still having problems.
